### PR TITLE
[SELC-3406] feat: Show in side menu button for invoiceable products

### PR DESCRIPTION
**Pull request by OraldoDoci**
#### List of Changes
Show Invoice button in side menu for products that are invoicable.
Update swagger of dashboard.

#### Motivation and Context
If a user has a role in a product thats is invoiceable the user should be able reach Invoice portal by clicking on the side menu button.

#### How Has This Been Tested?
Tested in dev enviroment

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/services/__mocks__/partyService.ts [hunk 1]
```diff
@@ -494,7 +495,6 @@ export const mockedParties: Array<Party> = [
         productId: 'prod-pn',
         authorized: true,
         productOnBoardingStatus: ProductOnBoardingStatusEnum.ACTIVE,
-        userRole: 'ADMIN',
```
Comment (loremedda) [inline src/services/__mocks__/partyService.ts:497]: Why removed userRole for this product?

File: src/services/__mocks__/partyService.ts [hunk 2]
```diff
@@ -319,7 +308,19 @@ export const mockedParties: Array<Party> = [
     ],
     vatNumber: '111122211111',
     // Use case with no onboarded products
-    products: [],
+    products: [
```
Comment (loremedda) [inline src/services/__mocks__/partyService.ts:311]: if you add a product, then remove the comment below

File: src/pages/dashboard/components/dashboardSideMenu/DashboardSideMenu.tsx [hunk 3]
```diff
@@ -105,6 +110,19 @@ export default function DashboardSideMenu({
               isPtPageVisible={false}
             />
           )}
+          {isInvoiceSectionVisible && (
+            <DashboardSidenavItem
+              title={'Fatturazione'}
```
Comment (loremedda) [inline src/pages/dashboard/components/dashboardSideMenu/DashboardSideMenu.tsx:115]: Localize this

**Review by loremedda — CHANGES_REQUESTED** [review]

File: src/services/__mocks__/partyService.ts (hunk 1 above)
Comment (OraldoDoci) [reply src/services/__mocks__/partyService.ts:497]: For tests.  The invoice button should not be visible if no userRole is present.

File: src/pages/dashboard/components/dashboardSideMenu/DashboardSideMenu.tsx (hunk 3 above)
Comment (OraldoDoci) [reply src/pages/dashboard/components/dashboardSideMenu/DashboardSideMenu.tsx]: done

File: src/services/__mocks__/partyService.ts (hunk 2 above)
Comment (OraldoDoci) [reply src/services/__mocks__/partyService.ts:310]: removed comment

**Comment by loremedda** [comment]
All resolved

**Review by manuraf — APPROVED** [review]

**Review by loremedda — APPROVED** [review]

**Pull request merged**